src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -421,6 +445,47 @@ public final SAMFileGATKReadWriter createSAMWriter(final File outputFile, final
     }
 
     /**
+     * Creates a VariantContextWriter who's outputFile type is determined by
+     * the outFile's extension, using the best available sequence dictionary for
+     * this tool, and default index, leniency and md5 generation settings.
+     *
+     * @param outFile output File for this writer. May not be null.
+     * @returns VariantContextWriter must be closed by the caller
+     */
+    public VariantContextWriter createVCFWriter(final File outFile) {
+        Utils.nonNull(outFile);
+
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+        Options options[];
+
+        if (createOutputVariantIndex) {
+            if (null == sequenceDictionary) {
+                logger.warn("An variant index will not be created - a sequence dictionary is required to create an output index");
+                // fall through and create without an index
+                options = lenientVCFProcessing ?
+                        new Options[]{Options.ALLOW_MISSING_FIELDS_IN_HEADER} :
```

Comment:
This logic is way more complicated than it needs to be.  Can you just use a 'List' instead of the '[]' and simply check each case and append or don't instead of repeating the ternary expressions for case?  Then just convert to array at the end.